colcon_bundle/verb/_overlay_utilities.py

```diff
@@ -47,19 +36,38 @@ def create_workspace_overlay(install_base: str,
         assets_directory,
         'v2_workspace_setup.sh'
     )
+    shellscript_dest = os.path.join(
+        workspace_staging_path,
+        'setup.sh'
+    )
     shellscript_path_bash = os.path.join(
         assets_directory,
         'v2_workspace_setup.bash'
     )
+    shellscript_dest_bash = os.path.join(
+        workspace_staging_path,
+        'setup.bash'
+    )
 
     # install_base: Directory with built artifacts from the workspace
     os.mkdir(workspace_staging_path)
-    shutil.copy2(shellscript_path,
-                 os.path.join(workspace_staging_path, 'setup.sh'))
-    os.chmod(os.path.join(workspace_staging_path, 'setup.sh'), 0o755)
-    shutil.copy2(shellscript_path_bash,
-                 os.path.join(workspace_staging_path, 'setup.bash'))
-    os.chmod(os.path.join(workspace_staging_path, 'setup.bash'), 0o755)
+    shutil.copy2(shellscript_path, shellscript_dest)
+    os.chmod(shellscript_dest, 0o755)
+
+    _generate_template(
```

Comment:
it looks like this is copying a file on line 54, just to overwrite it in the `_generate_template` step.